bson/src/main/org/bson/json/JsonReader.java

```diff
@@ -226,6 +226,8 @@ public BsonType readBsonType() {
         }
         if (noValueFound) {
             throw new JsonParseException("JSON reader was expecting a value but found '%s'.", token.getValue());
+        } else if (token.getType() == JsonTokenType.END_OF_FILE) {
```

Comment:
This seems to make a bad situation worse.  I see that this allows JsonReaderTest#testEndOfFile1 to pass, but is there any good reason why it _should_ pass?  The test subject is

    "{ test : "

so why should JsonReader generate BsonType.END_OF_DOCUMENT when there is no closing curly brace?